Lib/concurrent/futures/_base.py

```diff
@@ -594,10 +599,21 @@ def map(self, fn, *iterables, timeout=None, chunksize=1):
                 before the given timeout.
             Exception: If fn(*args) raises for any values.
         """
+        if buffersize is not None and buffersize < 1:
+            raise ValueError("buffersize must be None or >= 1.")
+
         if timeout is not None:
             end_time = timeout + time.monotonic()
 
-        fs = [self.submit(fn, *args) for args in zip(*iterables)]
+        args_iter = iter(zip(*iterables))
+        if buffersize:
+            fs = collections.deque(
+                self.submit(fn, *args) for args in islice(args_iter, buffersize)
```

Comment:
Isn't buffersize empty? Can you introduce it? (Forgive me for not understanding it).

Reply:
Hey @rruuaanng , I have reworked the PR's description, I hope it makes things clearer!